content/stable-typescript-types-in-ember-5-1.md

```diff
@@ -0,0 +1,359 @@
+---
+title: Stable TypeScript Types in Ember 5.1 
+authors:
+  - chris-krycho
+  - the-ember-typescript-team
+date: 2023-06-26T00:00:00.000Z
+tags:
+  - '2023'
+  - typescript
+  - announcement
+  - version-5-x
+---
+
+The headlining feature of Ember 5.1 is stable [TypeScript](https://www.typescriptlang.org) support! 🎉 The framework now publishes TypeScript types built from Ember's own source code, with Ember's [strong Semantic Versioning commitment](https://emberjs.com/releases/) baked in—bringing benefits for both TypeScript and JavaScript users. This dedicated blog post walks through all the details.
+
+- [Getting started](#toc_getting-started)
+    - [JavaScript](#toc_javascript)
+    - [TypeScript](#toc_typescript)
+- [Caveats](#toc_caveats)
+- [Stability](#toc_stability)
+- [Migration Guide](#toc_migration-guide)
+    - [From the preview types](#toc_from-the-preview-types)
+    - [From DefinitelyTyped](#toc_from-definitelytyped)
+- [Background](#toc_background)
+- [Onward](#toc_onward)
+
+
+## Getting started
+
+Both JavaScript and TypeScript developers can benefit from using these new type definitions. If you are JavaScript developer, this will make automatic imports, in-editor framework documentation, and other features work much better for you in your Ember app or addon. If you are a TypeScript user, you will get all of those benefits as well as the strong type safety guarantees and refactoring capabilities you are used to.
+
+### JavaScript
+
+In your `app/app.js`, add this single doc comment to the bottom of the file:
+
+```js
+/**
+ * @typedef {import('ember-source/types')} EmberTypes
+ */
+```
+
+Now you will get autocomplete and in-editor documentation for all the `@ember/*` packages which ship with Ember with no other adjustment required!
```

Comment:
Is this true regardless of IDE or only for IDEs like VSCode that come with baked-in TS support?

Reply:
Ah! Good catch. We should clarify that it will be true *for any editor configured to use the TS LS*. I will update to note that on Monday!

Reply:
```suggestion
This is all that is necessary to unlock autocomplete and in-editor documentation for all the `@ember/*` packages which ship with Ember with no other adjustment required! Any editor configured to use the TypeScript language server for JavaScript support will "just work" once you make this change—including VS Code, where this works in a default installation.
```